### PR TITLE
[ML] retrieve real memory usage of pytorch inference

### PR DESCRIPTION
**Pull request by hendrikmuhs**
add functionality to get the real memory usage and integrate it into pytorch inference

### CODE REVIEW

File: lib/core/CProcessStats_Linux.cc
Comment (droberts195) [inline lib/core/CProcessStats_Linux.cc:63]: `stol` can throw an exception, and our code is written to avoid exceptions as much as possible.

I think it's safer to use `CStringUtils::stringToType` and log an error and return 0 if that fails. Otherwise this code is relying on the format of `/proc/self/statm` never changing and if it does change then we might find our process crashing just because it couldn't get memory info.

Also, please check that at least 2 tokens were found before accessing `tokens[1]`.

File: docs/CHANGELOG.asciidoc
Comment (droberts195) [inline docs/CHANGELOG.asciidoc:35]: On it's own this isn't a user-facing change. There will need to be a Java PR to make this do anything as far as end users are concerned. So I would add the release note for the overall functionality in the Java PR. The C++ release notes have to be manually copied into the overall release notes, whereas the Java release notes are auto-generated, so less work.

Comment (hendrikmuhs) [reply docs/CHANGELOG.asciidoc:35]: I made this entry, because of the github workflow check. I happily revert and ignore the gh check. 

Maybe it can be documented in the process that this check isn't mandatory to pass.

Comment (droberts195) [reply docs/CHANGELOG.asciidoc:35]: If you change `>enhancement` to `>non-issue` then a release note won't be expected. Then for the Java PR set the labels the other way around.

File: lib/core/CProcessStats_Linux.cc
Comment (droberts195) [inline lib/core/CProcessStats_Linux.cc:74]:
```suggestion
            return 0;
        }
```

(Or alternatively intialize `rss` to 0 on line 59.)

File: lib/core/CProcessStats_Linux.cc
Comment (droberts195) [inline lib/core/CProcessStats_Linux.cc:84]:
```suggestion
        LOG_DEBUG(<< "failed to get resource usage(getrusage): " << ::strerror(errno));
```

File: lib/core/CProcessStats_MacOSX.cc
Comment (droberts195) [inline lib/core/CProcessStats_MacOSX.cc:31]:
```suggestion
        LOG_DEBUG(<< "failed to get resource usage(getrusage): " << ::strerror(errno));
```

**Review by droberts195 — APPROVED** [review]
LGTM if you could just fix 3 minor nits before merging

**Pull request merged**
